runtime/centrifuge/src/migrations.rs

```diff
@@ -9,5 +9,126 @@
+	#[cfg(feature = "try-runtime")]
+	const LP_ETH_USDC: CurrencyId = CurrencyId::ForeignAsset(100_001);
+	const DOT_NATIVE_USDC: CurrencyId = CurrencyId::ForeignAsset(6);
+
+	pub struct Migration;
+
+	impl OnRuntimeUpgrade for Migration {
+		#[cfg(feature = "try-runtime")]
+		fn pre_upgrade() -> Result<Vec<u8>, &'static str> {
+			let pool_details: PoolDetailsOf<Runtime> =
+				PoolSystem::pool(ANEMOY_POOL_ID).ok_or("Could not find Anemoy Pool")?;
+
+			ensure!(
+				pool_details.currency == LP_ETH_USDC,
+				"anemoy_pool::Migration: pre_upgrade failing as Anemoy's currency should be LpEthUSDC"
+			);
+
+			Ok(pool_details.encode())
+		}
+
+		fn on_runtime_upgrade() -> Weight {
+			// To be executed at 1021, reject higher spec_versions
+			if crate::VERSION.spec_version >= 1022 {
+				log::error!(
+					"anemoy_pool::Migration: NOT execution since VERSION.spec_version >= 1022"
+				);
+				return Weight::zero();
+			}
+
+			let (sanity_checks, weight) = verify_sanity_checks();
+			if !sanity_checks {
+				log::error!("anemoy_pool::Migration: Sanity checks FAILED");
+				return weight;
+			}
+
+			pallet_pool_system::Pool::<Runtime>::mutate(ANEMOY_POOL_ID, |details| {
+				let details = details.as_mut().unwrap();
+				details.currency = DOT_NATIVE_USDC;
+				log::info!("anemoy_pool::Migration: currency set to USDC ✓");
+			});
+
+			weight.saturating_add(
+				<Runtime as frame_system::Config>::DbWeight::get().reads(
+					pallet_pool_system::Pool::<Runtime>::iter_keys()
+						.count()
+						.saturating_mul(2) as u64,
+				),
+			)
+		}
+
+		#[cfg(feature = "try-runtime")]
+		fn post_upgrade(old_state: Vec<u8>) -> Result<(), &'static str> {
+			let mut old_pool_details = PoolDetailsOf::<Runtime>::decode(&mut old_state.as_ref())
+				.map_err(|_| "Error decoding pre-upgrade state")?;
+
+			let pool_details: PoolDetailsOf<Runtime> =
+				PoolSystem::pool(ANEMOY_POOL_ID).ok_or("Could not find Anemoy Pool")?;
+
+			// Ensure the currency set to USDC is the only mutation performed
+			old_pool_details.currency = DOT_NATIVE_USDC;
+			ensure!(
+				old_pool_details == pool_details,
+				"Corrupted migration: Only the currency of the Anemoy pool should have changed"
+			);
+
+			log::info!("anemoy_pool::Migration: post_upgrade succeeded ✓");
+			Ok(())
+		}
+	}
+
+	fn verify_sanity_checks() -> (bool, Weight) {
+		let res =
+			pallet_investments::ActiveInvestOrders::<Runtime>::iter_keys()
+				.filter(|investment| investment.pool_id == ANEMOY_POOL_ID)
+				.count() == 0 && pallet_investments::ActiveInvestOrders::<Runtime>::iter_keys()
+				.filter(|investment| investment.pool_id == ANEMOY_POOL_ID)
+				.count() == 0 && pallet_investments::InvestOrders::<Runtime>::iter_keys()
+				.filter(|(_, investment)| investment.pool_id == ANEMOY_POOL_ID)
+				.count() == 0 && pallet_investments::RedeemOrders::<Runtime>::iter_keys()
+				.filter(|(_, investment)| investment.pool_id == ANEMOY_POOL_ID)
+				.count() == 0;
+
```

Comment:
That looks soo clean! nice!

Reply:
thanks 🙏 just pushed the check for pool value == 0 and realised that we were checking `ActiveInvestOrders` twice instead of checking `ActiveRedeemOrders` :) fixed